File: CS/DXSample/ViewModel/ProductsViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DXSample.Common;

namespace DXSample.ViewModel {
    public class ProductsViewModel {
        public virtual string Content { get; set; }
        protected ProductsViewModel() {
            Content = "A representation of ProductsViewModel.";
            ViewInjectionManager.Default.RegisterNavigatedEventHandler(this, () => {
                ViewInjectionManager.Default.Navigate(Regions.Navigation, ModuleType.Products);
            });
        }
        public static ProductsViewModel Create() {
            return ViewModelSource.Create(() => new ProductsViewModel());
        }
    }
}

File: VB/DXSample/ViewModel/ProductsViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DXSample.Common

Namespace DXSample.ViewModel

    Public Class ProductsViewModel

        Public Overridable Property Content As String

        Protected Sub New()
            Content = "A representation of ProductsViewModel."
            ViewInjectionManager.Default.RegisterNavigatedEventHandler(Me, Sub() ViewInjectionManager.Default.Navigate(Navigation, ModuleType.Products))
        End Sub

        Public Shared Function Create() As ProductsViewModel
            Return ViewModelSource.Create(Function() New ProductsViewModel())
        End Function
    End Class
End Namespace
